ipn/local.go

```diff
@@ -333,6 +358,42 @@ func (b *LocalBackend) popBrowserAuthNow() {
 	}
 }
 
+func (b *LocalBackend) loadStateWithLock(key StateKey, prefs *Prefs) error {
+	switch {
+	case key != "" && prefs != nil:
+		b.logf("Importing frontend prefs into backend store")
+		if err := b.store.WriteState(key, prefs.ToBytes()); err != nil {
+			return fmt.Errorf("store.WriteState: %v", err)
+		}
+		fallthrough
```

Comment:
fallthrough is used so rarely in Go that even when it's the right thing to use, I try to avoid it so readers don't need to think what it does (as they're just not used to seeing it and might not know what it means, or quickly know whether it means the next case is evaluated or not)

Could you move the panic up first, and then just use some if statements instead of the switch?

```go
    if key == "" {
          if prefs == nil {
                 panic("state key and prefs are both unset")
          }
          b.logf("Using frontend prefs")
          b.prefs = *prefs
          b.stateKey = ""
          return nil
    }
    if prefs != nil {
           // migrate
    }
    // Using backend prefs
```

then slightly less indented too for the bulk of the code

Reply:
Adjusted as you suggest.

In general, I like `switch` for this kind of complex decision making, because I feel like it's a clearly signposted "Decision time! One of these things will happen." But as you point out, using `fallthrough` means suddenly several of these things may happen, and at that point we want more "basic" control flow.